pdstable/pds4table.py

```diff
@@ -0,0 +1,331 @@
+            try:
+                self.table_file_name = file_area['File']['file_name']
+                self.table_file_li = [self.table_file_name]
+            except:
+                raise IOError('Table file name was not found in PDS4 label')
+
+            if table_file is not None and table_file not in self.table_file_li:
+                raise ValueError("The provided table file name doesn't match the one" +
+                                 'in the label. The label contains one table file ' +
+                                 f'{self.table_file_name}')
+        # The label file points to multiple table files
+        elif isinstance(file_area, list):
+            try:
+                table_name_li = [f['File']['file_name'] for f in file_area]
+            except:
+                raise IOError('Table file name was not found in PDS4 label')
+
+            if table_file is None or table_file not in table_name_li:
+                raise ValueError(f"The table file name '{table_file}' doesn't exist. " +
+                                 f'The label contains {len(table_name_li)} table ' +
+                                 f'files: {table_name_li}')
+            else:
+                self.table_file_name = table_file
+                self.table_file_li = table_name_li
+                # specify the table file that we want to read
+                idx = table_name_li.index(table_file)
+                file_area = file_area[idx]
+        # The label file has no table file info
+        else:
+            raise ValueError(f'{label_file_path} does not contain any table file info.' )
+
+        try:
+            self.header_bytes = int(file_area['Header']['object_length'])
+        except KeyError:
+            # Some tables don't have header
+            self.header_bytes = 0
+
+        # table info is the last child tag of the file area
+        table_area = file_area[list(file_area.keys())[-1]]
+        self.rows = int(table_area['records'])
+        # record info is the last child tag of the table info area
+        record_area = table_area[list(table_area.keys())[-1]]
+        self.columns = int(record_area['fields'])
+
+        try:
+            # for the table with fixed row length
+            self.row_bytes = int(record_area['record_length'])
+            self.fixed_length_row = True
+            self.field_delimiter = None
+        except:
+            # for the case like .csv table, row length is not used
+            self.row_bytes = int(record_area['maximum_record_length'])
+            self.fixed_length_row = False
+            self.field_delimiter = PDS4_FIELD_DELIMITER[table_area['field_delimiter']]
+
+        # Save the key info about each column in a list and a dictionary
+        self.column_info_list = []
+        self.column_info_dict = {}
+
+        # Construct the dtype0 dictionary
+        self.dtype0 = {'crlf': ('|S2', self.row_bytes-2)}
+
+        default_invalid = set(invalid.get('default', []))
```

Comment:
You should ask @markshowalter if we even want `invalid` for PDS4. The table descriptions should include the invalid constants in the table records. See https://pds.nasa.gov/datastandards/documents/im/current/index_1N00.html#class_pds_special_constants

This argument is present because the PDS3 labels are so bad and often miss things like the invalid constant in the label so we have to supply it while reading the label. But the PDS4 labels should be better than that. And you need to get the invalid constant from the label record.